core/src/main/java/io/github/tcdl/msb/Consumer.java

```diff
@@ -96,12 +99,14 @@ protected void handleRawMessage(String jsonMessage) {
             LOG.debug("Message has been successfully parsed {}", jsonMessage);
 
             if (!isMessageExpired(message)) {
-                messageHandler.handleMessage(message);
+                messageHandler.handleMessage(message, acknowledgeHandler);
             } else {
                 LOG.warn("Expired message: {}", jsonMessage);
+                acknowledgeHandler.rejectMessage();
```

Comment:
When reading this line it's not immediately obvious whether `acknowledgeHandler.rejectMessage();` requeues the message or not. And it's critical here that it should not. In my opinion the method name is not expressive enough here.